src/ai/backend/storage/weka/weka_client.py

```diff
@@ -0,0 +1,383 @@
+import json
+import logging
+import time
+import urllib.parse
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import aiohttp
+from aiohttp import web
+
+from ai.backend.common.logging import BraceStyleAdapter
+from ai.backend.common.types import BinarySize
+
+from .exceptions import WekaNotFoundError, WekaUnauthorizedError
+
+log = BraceStyleAdapter(logging.getLogger(__name__))
+
+
+@dataclass
+class WekaQuota:
+    quota_id: str
+    inode_id: int
+    hard_limit: Optional[int]
+    soft_limit: Optional[int]
+    grace_seconds: Optional[int]
+
+    @classmethod
+    def from_json(cls, quota_id: str, data: Any):
+        return WekaQuota(
+            quota_id,
+            data["inode_id"],
+            data["hard_limit_bytes"],
+            data["soft_limit_bytes"],
+            data["grace_seconds"],
+        )
+
+    def to_json(self):
+        return {
+            "quota_id": self.quota_id,
+            "inode_id": self.inode_id,
+            "hard_limit": self.hard_limit,
+            "soft_limit": self.soft_limit,
+            "grace_seconds": self.grace_seconds,
+        }
+
+
+@dataclass
+class WekaFs:
+    id: str
+    name: str
+    uid: str
+    group_id: str
+    group_name: str
+    used_total_data: int
+    used_total: int
+    free_total: int
+    available_total: int
+    available_ssd_metadata: int
+    used_ssd: int
+    free_ssd: int
+    available_ssd: int
+    ssd_budget: int
+    total_budget: int
+
+    def from_json(data: Any):
+        return WekaFs(
+            data["id"],
+            data["name"],
+            data["uid"],
+            data["group_id"],
+            data["group_name"],
+            data["used_total_data"],
+            data["used_total"],
+            data["free_total"],
+            data["available_total"],
+            data["available_ssd_metadata"],
+            data["used_ssd"],
+            data["free_ssd"],
+            data["available_ssd"],
+            data["ssd_budget"],
+            data["total_budget"],
+        )
+
+
+def error_handler(inner):
+    async def outer(*args, **kwargs):
+        try:
+            return await inner(*args, **kwargs)
+        except web.HTTPNotFound:
+            raise WekaNotFoundError
```

Comment:
Wouldn't it better to handle other HTTP errors and wrap them as `WekaAPIError`?

Reply:
Hm, please ignore the above comment. Since this is a server-side code, so I think we don't need to mixin `aiohttp.HTTPException` classes here but we need to wrap these internal exceptions in the manager-facing or client-facing **API handlers**.